### ISSUE TITLE
training epoch size

### ISSUE THREAD

**Issue by bragilee** [ClementPinard/FlowNetPytorch#43]
Hi,

Thank you so much for your work.

One thing I am a bit confused about the training is what is the "epoch size"? Is it the iterations of each epoch?

Thank you so much. :)

**Comment by ClementPinard** [comment]
It is indeed the number of iterations of each epoch. It means that the number of samples seen during training is both a function of epoch size and batch size.

I took this choice because that way when changing batch size you only the effect of mini batch training without the effect on epoch : you have the same number of gradient descents and then can compare directly with a lower batchsize

**Comment by bragilee** [comment]
Hi,

Thank you so much for your detailed explanations.

One thing is in this project, the epoch size is 1000, which means in each epoch, there will be 8*1000 samples used for training, not all training samples are used, is that right?

**Comment by ClementPinard** [comment]
It's right, the epoch is more about the number of iteration between two validation than a complete loop of the dataset. In fact when the code was released (like 1 year ago), random data loading was truly random, and not a random permutation of your data, which means in the same epoch you could see a sample multiple times, and the epoch did not have reason to be linked to dataset size. Now it's not true anymore, but I actually find it more practical to link it to your number of iterations.

**Comment by bragilee** [comment]
Got it.
Thank you so much.
